build.sbt

```diff
@@ -342,6 +339,17 @@ lazy val compilerInterface = (project in internalPath / "compiler-interface")
   )
   .configure(addSbtUtilInterface)
 
+/* Create a duplicated compiler-interface project that uses Scala 2.10 to parse Java files.
+ * Scala 2.10's parser uses Java 6 semantics, so this way we ensure that the interface can
+ * be compiled with Java 6 too. `compiler-interface` checks compilation for Java 8. */
+val compilerInterfaceJava6Compat = compilerInterface
+  .withId("compilerInterfaceJava6Compat")
+  .settings(
+    scalaVersion := scala210,
+    crossScalaVersions := Seq(scala210),
+    target := (target in compilerInterface).value / "java6-parser-compat"
```

Comment:
Should we add `skip in publish := true`?

Reply:
Indeed!